pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1254,6 +1254,32 @@ static __always_inline bool is_compat(struct task_struct *task)
 #endif
 }
 
+static __always_inline int get_syscall_id_from_regs(struct pt_regs *regs)
+{
+#if defined(bpf_target_x86)
+    int id = READ_KERN(regs->orig_ax);
+#elif defined(bpf_target_arm64)
+    int id = READ_KERN(regs->syscallno);
+#endif
+    return id;
+}
+
+static __always_inline struct pt_regs *get_task_pt_regs(struct task_struct *task)
+{
+#if defined(bpf_target_x86)
+    void *__ptr = READ_KERN(task->stack) + THREAD_SIZE - TOP_OF_KERNEL_STACK_PADDING;
+    return ((struct pt_regs *) __ptr) - 1;
+#elif defined(bpf_target_arm64)
+    return ((struct pt_regs *) (THREAD_SIZE + READ_KERN(task->stack)) - 1);
```

Comment:
I think this won't compile with arm64 CO-RE as THREAD_SIZE is not defined in missing_definitions.h

Reply:
You're right it doesn't. Working backwards through the ARM definitions right now (with help from @rafaeldtinoco).